src/hooks/usePools.ts

```diff
@@ -37,7 +37,7 @@ export function usePools(
     const proAmmCoreFactoryAddress = chainId && NETWORKS_INFO[chainId].elastic.coreFactory
 
     return transformed.map(value => {
-      if (!proAmmCoreFactoryAddress || !value) return undefined
+      if (!proAmmCoreFactoryAddress || !value || !value[0].equals(value[1])) return undefined
```

Comment:
if `value = []` it still crash.

Reply:
@namgold

Reply:
replace to !value[0]?.equals(value[1]) is enough

Reply:
`value` will never be `[]`

Reply:
Look at [here](https://github.com/KyberNetwork/kyberswap-interface/blob/cc37a3e9d4ce5bb0605e7fe79a47f49c68657de1/src/hooks/usePools.ts#L33), it's right above this line